blog/2024/10-26-media-segments-10.10.0/index.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: Jellyfin Media Segments Explained 10.10.0
+description: What are Media Segments and how to use them in version 10.10.0
+authors: JPVenson
+slug: jellyfin-media-segments-10.10.0
+tags: [release]
+---
+
+First of all things i would like to introduce myself as a new Server-Team member for the jellyfin FOSS project.
+
+I have been active for a bit now, first just as an Freelance software developer that created its own small media server because i did not wanted to pay for other serivces but then quickly learned what it all takes.
+
+After setting up Jellyfin i stuck around in the chat and later started to help out as a community moderator, while submitting some features and bugfixes. However with my last PR implementing Media Segments, i joined the Server Team to bring jellyfin new features (and some bugs ;-)).
+
+\- JPVenson
+
+{/* truncate */}
+
+## Media Segments Overview
+
+First lets start with the Basics of what a Media Segment is.
+A Media Segment exists only within our Database and defines a timespan within a Movie or Episode with a certain types.
+
+Those types can be currently one of the following:
+
+- Commercial
+- Preview
+- Recap
+- Outro
+- Intro
+
+### An example
+
+An episode that is 16 minutes (00:16:00) long, can have multiple segments:
+
+1. from 00:00:00 until 00:03:08 there is an `Intro`
+2. from 00:08:03 until 00:08:59 there is an `Commerical`
+3. from 00:14:30 until 00:16:00 there is an `Outro`
+
+Jellyfin can now store this information and provide it via the new MediaSegmentApi to clients. 
+What we did not implement in 10.10 however, is the creation of those Segments. 
+Jellyfin only stores and provides this data to clients but to generate it you need a Plugin that provide this data to jellyfin to be stored.
+
+We did create a plugin to create segments based on Chapter names which sounds similar but is something different. 
+A Chapter is just any section of a video, but without any defined type. 
+A MediaSegment on the other hand is a section in a video, with a defined type.
+
+## Media Segment Plugins
+
+Media Segment plugins can now utilise our new System to store their information about Intros/Outros/Commericals and all other types of segments. 
+This information is then used by clients via our Api to provide actions like a "Skip" button in their UI.
+The big benefit here is that this approach generalises how Segments are handled so gone are the days where you needed to inject javascript into the webUI to get IntroSkipping.
+
+This does *NOT* make the existing plugins obsolete however as we still rely on them to generate the segments in the first place. 
+But as more clients implement support, we can bring their Segments to more clients and provide for exmaple IntroSkipping.
+
+## Media Segments on clients
+
+Currently the WebUI and the AndroidTV client support MediaSegments. 
```

Comment:
MediaSegments for the AndroidTV client has only been implemented in the Master branch and is not available in the current release.